src/realm/table.cpp

```diff
@@ -1060,14 +1060,23 @@ ColKey Table::do_insert_root_column(ColKey col_key, ColumnType type, StringData
         m_tombstones->insert_column(col_key);
     }
     // create string interners internal rep as well as data area
+
     REALM_ASSERT_DEBUG(m_interner_data.is_attached());
     while (col_ndx >= m_string_interners.size()) {
         m_string_interners.push_back({});
     }
+
+    if (col_key.get_type() != col_type_String) {
```

Comment:
This is actual a better tentative fix for the problem. It seems, we are adding a `string interner` for every `ColKey`, despite the fact that the column may not be a string. Am I missing something here?
I think, this is a problem (perf aside), we are exposing the interner via the ClusterIterator. I am not sure why this is happening, it is just empirical evidence following the audit test.
But in general, shouldn't we just have an interner for strings if the column is a string? ...

Reply:
Not quite. We want interners for all the leafs which may hold strings, so it also includes Mixed and anything which can include Mixed (so dicts and lists, I guess)

Having an interner for each column is wasteful, but WHY does it cause failures? I'd prefer the code to be so robust, that adding an interner which is never used still behaves correctly.